### PR TITLE
#10124: Move non_zero_indices op

### PR DESCRIPTION
**Pull request by KalaivaniMCW**
### Ticket
Link to Github Issue #10124 

### Problem description
Migrate non_zero_indices op

### What's changed
Migrated non_zero_indices op to ttnn/operations/data_movement

### Checklist
- [ ] Post commit CI  https://github.com/tenstorrent/tt-metal/actions/runs/10303386055
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: ttnn/cpp/ttnn/operations/data_movement/non_zero_indices/non_zero_indices.hpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <vector>
+
+#include "ttnn/run_operation.hpp"
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/data_movement/non_zero_indices/non_zero_indices.hpp:9]: Please remove this header, it is not needed and is heavy

File: ttnn/cpp/ttnn/operations/data_movement/non_zero_indices/non_zero_indices_pybind.hpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "ttnn/cpp/pybind11/decorators.hpp"
+
+#include "non_zero_indices.hpp"
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/data_movement/non_zero_indices/non_zero_indices_pybind.hpp:12]: please split to hpp and cpp
best example is ccl/all_gather
it also shows how to use pybind_fwd.hpp to not include anything heavy in this header

**Review by ayerofieiev-tt — APPROVED** [review]
Left some comments. Please address them before merging.
Thank you!

**Review by eyonland — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/data_movement/non_zero_indices/non_zero_indices_pybind.hpp (hunk 2 above)
Comment (KalaivaniMCW) [reply ttnn/cpp/ttnn/operations/data_movement/non_zero_indices/non_zero_indices_pybind.hpp]: done

File: ttnn/cpp/ttnn/operations/data_movement/non_zero_indices/non_zero_indices.hpp (hunk 1 above)
Comment (KalaivaniMCW) [reply ttnn/cpp/ttnn/operations/data_movement/non_zero_indices/non_zero_indices.hpp]: removed. Thanks

**Review by sjameelTT — APPROVED** [review]

**Review by ntarafdar — APPROVED** [review]

**Pull request merged**